sdk/pinocchio/src/cpi.rs

```diff
@@ -92,58 +129,80 @@ pub fn invoke_signed<const ACCOUNTS: usize>(
 /// Invoke a cross-program instruction with signatures from a slice of
 /// `AccountInfo`s.
 ///
+/// This function performs validation of the `account_infos` slice to ensure that:
+///   1. The accounts match the expected accounts in the instruction, i.e., their
+///      `Pubkey` matches the `pubkey` in the `AccountMeta`.
+///   2. The borrow state of the accounts is compatible with the mutability of the
+///      accounts in the instruction.
+///
+/// This validation is done to ensure that the borrow checker rules are followed,
+/// consuming CUs in the process. The `invoke_signed_unchecked` is an alternative
+/// to this function that have lower CU consumption since it does not perform
+/// any validation. This should only be used when the caller is sure that the borrow
+/// checker rules are followed.
+///
 /// # Important
 ///
 /// The accounts on the `account_infos` slice must be in the same order as the
-/// `accounts` field of the `instruction`.
-#[inline]
+/// `accounts` field of the `instruction`. When the instruction has duplicated
+/// accounts, it is necessary to pass a duplicated reference to the same account
+/// to maintain the 1:1 relationship between `account_infos` and `accounts`.
 pub fn slice_invoke_signed(
     instruction: &Instruction,
     account_infos: &[&AccountInfo],
     signers_seeds: &[Signer],
 ) -> ProgramResult {
-    if instruction.accounts.len() < account_infos.len() {
-        return Err(ProgramError::NotEnoughAccountKeys);
-    }
-
-    if account_infos.len() > MAX_CPI_ACCOUNTS {
+    // Check that the number of accounts match the expected accounts by
+    // the instruction and do not exceed the maximum allowed.
+    if instruction.accounts.len() != account_infos.len() || account_infos.len() > MAX_CPI_ACCOUNTS {
```

Comment:
Should there be the same check in `invoke_signed()`?
Or are you still trying to figure out the CU increase it is causing?

P.S.  Is a 100 CU increase a substantially bad thing?
Just curious as to the scale of the CU costs.

Reply:
A `100` increase for a CPI call is not good since it represents almost half of the variable CU required for a CPI.